src/main/java/com/github/manolo8/darkbot/core/api/NativeApiAdapter.java

```diff
@@ -12,8 +13,8 @@
     private static int nextBotId = 0;
     private int botId = -1;
 
-    public NativeApiAdapter() {
-        this.loginData = LoginUtils.performUserLogin();
+    public NativeApiAdapter(StartupParams params) {
+        this.loginData = params.shouldAutoLogin() ? LoginUtils.performAutoLogin(params) : LoginUtils.performUserLogin();
```

Comment:
```suggestion
        this.loginData = LoginUtils.performUserLogin(params);
```
pass the params and inside the performUserLogin, do 
`if (params != null && params.shouldAutoLogin()) return LoginUtils.performAutoLogin(params)`
It will simplify having this logic in 3 different api adapters, and will put it in one place. Also allow for params to be null.